mmtk/src/util.rs

```diff
@@ -152,3 +156,159 @@ pub extern "C" fn mmtk_get_possibly_forwared(object: ObjectReference) -> ObjectR
         None => object,
     }
 }
+
+// Functions to set the side metadata for the VO bit (copied from mmtk-core)
+pub fn bulk_update_vo_bit(
+    start: Address,
+    size: usize,
+    update_meta_bits: &impl Fn(Address, u8, Address, u8),
+) {
+    // Update bits for a contiguous side metadata spec. We can simply calculate the data end address, and
+    // calculate the metadata address for the data end.
+    let update_contiguous = |data_start: Address, data_bytes: usize| {
+        if data_bytes == 0 {
+            return;
+        }
+        let meta_start = address_to_meta_address(data_start);
+        let meta_start_shift = meta_byte_lshift(data_start);
+        let meta_end = address_to_meta_address(data_start + data_bytes);
+        let meta_end_shift = meta_byte_lshift(data_start + data_bytes);
+        update_meta_bits(meta_start, meta_start_shift, meta_end, meta_end_shift);
+    };
+
+    // VO bit is global
+    update_contiguous(start, size);
+}
+
+/// Performs the translation of data address (`data_addr`) to metadata address for the specified metadata (`metadata_spec`).
+pub fn address_to_meta_address(data_addr: Address) -> Address {
+    #[cfg(target_pointer_width = "32")]
+    let res = {
+        if metadata_spec.is_global {
+            address_to_contiguous_meta_address(metadata_spec, data_addr)
+        } else {
+            address_to_chunked_meta_address(metadata_spec, data_addr)
+        }
+    };
+    #[cfg(target_pointer_width = "64")]
+    let res = { address_to_contiguous_meta_address(data_addr) };
+
+    res
+}
+
+/// Performs address translation in contiguous metadata spaces (e.g. global and policy-specific in 64-bits, and global in 32-bits)
+pub fn address_to_contiguous_meta_address(data_addr: Address) -> Address {
+    let log_bits_num = 0 as i32; // VO_BIT log_num_of_bits
```

Comment:
You can create constants for  `VO_BIT_LOG_NUM_OF_BITS` and `VO_BIT_LOG_BYTES_PER_REGION`.